### ISSUE TITLE
Issue with row values from "Related Object" of IQueryable

### ISSUE THREAD

**Issue by jjkcharles** [NonFactors/MVC6.Grid#26]
If I pass in a IQueryable type which includes a related object (that is used to display a column's value), the data of the particular Column is lost when applying filters or doing a paging request.

Lets say, I have a Grid, which passes in "Posts" object which has "Comments" object has a related object. Represented as below,

`IQueryable<Post> gridData = _dbContext.Posts.Include(p=>p.Comments)`

And one of the columns displays the number of comments in a post, whose expression is as below

`model => model.Comments.Count()`
_(This column is correctly rendered for the first page of result, but on paging/filtering, it will be displayed as 0)_

**Note**: when applying filters, the first page seems to be showing correct data (which is strange! either it was a random coincidence or I am missing something)

**Comment by Muchiachio** [comment]
Could you share a repro for this? I tried it, but couldn't repro the count being zero on consecutive calls.

**Comment by jjkcharles** [comment]
Here you go, https://github.com/jjkcharles/SampleMVCGrid

**Comment by Muchiachio** [comment]
I ran it, and it didn't lost comment count neither on paging nor on filtering, here's what I get

| Post | Comment count |
|------|---:|
| Post01 | 3 |
| Post02 | 2 |
| Post03 | 1 |
| Post04 | 1 |
| Post05 | 1 |
| Post06 | 1 |
| Post07 | 1 |
| Post08 | 1 |
| Post09 | 1 |
| Post10 | 3 |
| Post11 | 0 |
| Post12 | 0 |
| Post13 | 0 |
| Post14 | 0 |
| Post15 | 0 |
| Post16 | 0 |
| Post17 | 0 |
| Post18 | 0 |

**Comment by jjkcharles** [comment]
Try filtering by "No. of Comments" column with condition ">0" (ideally, any filtering should do), then try to go to the second page (where No. of Comments column will be "0").

**Comment by Muchiachio** [comment]
Ok, I got zeros.

**Comment by Muchiachio** [comment]
This seems to be an [EntityFramework](https://github.com/aspnet/EntityFramework) issue, which was already reported in the past and should be fixed with 1.0.2 or 1.1.0 releases.

You can see this by running this query in your controller:
```C#
var pagedPosts = posts.Where(post => post.Comments.Count > 0).Skip(4).Take(4).ToList();
```

**Comment by jjkcharles** [comment]
Thanks for checking on this. I did update the Entity Framework to Preview 1.1.0, and it seems to be working fine.
